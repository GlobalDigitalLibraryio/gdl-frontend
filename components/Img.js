// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 * 
 * See LICENSE
 */

import styled from 'styled-components';
import * as React from 'react';

const Img = styled.img`
  max-height: 100%;
  max-width: 100%;
`;

const NO_COVER_PLACEHOLDER_URL = '/static/placeholder-cover.png';

/**
 * Use for book covers. Renders a placeholder if no image is passed
 */
export default ({ src, alt }: { src: ?string, alt: string }) => (
  <Img src={src || NO_COVER_PLACEHOLDER_URL} alt={alt} aria-hidden />
);
