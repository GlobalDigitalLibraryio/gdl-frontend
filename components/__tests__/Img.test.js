// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import Img from '../Img';

test('Renders', () => {
  // src here is a 1x1 black gif
  const tree = mount(
    <Img
      alt="test"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs="
    />,
  );

  expect(toJson(tree)).toMatchSnapshot();
});
