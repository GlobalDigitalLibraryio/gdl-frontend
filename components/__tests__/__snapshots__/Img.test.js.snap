// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
.c0 {
  max-height: 100%;
  max-width: 100%;
}

<Component
  alt="test"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs="
>
  <styled.img
    alt="test"
    aria-hidden={true}
    src="data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs="
  >
    <img
      alt="test"
      aria-hidden={true}
      className="c0"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs="
    />
  </styled.img>
</Component>
`;
